libraries/TeamSpeak3/Helper/String.php

```diff
@@ -390,7 +390,7 @@ public function filterDigits()
    */
   public function isInt()
   {
-    return (is_numeric($this->string) && !$this->contains(".") && !$this->contains("x")) ? TRUE : FALSE;
+      return ctype_digit($this->string);
```

Comment:
I'm still worried this might break something and the recent revert from @svenpaulsen makes me hesitant to pass this through.

The issue here is that `isInt()` is not used to check for type `int` but rather more to check if string is numeric, like native `is_numeric()` and the name of this should probably be changed to `isNumeric()`.
This means, things we're testing for such as `(string) "+1"`, `(string) "-1"`, `(string) "1"` actually are valid numbers and should be cast to `int` as the framework expects them to be.
If we change this to `ctype_digit()`, channel names and other things that should be strings will work correctly, but other logic expecting `int` values will instead get things like `(string) "1"` and that may cause issues.
I personally think we should do some smarter type checking / validation when parsing server reply to rule out these cases.

I'm working on some integration testing in https://github.com/ronindesign/ts3phpframework/tree/testing so we can know if there are cases where this might break.